### PR TITLE
fix(bpmn-model): add validation for blank formId and formKey

### LINKED ISSUE #14370: Check for empty formId and formKey during process deploy

### PR DESCRIPTION
**Pull request by nicpuppa**
## Related issues

closes #14370 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebeElementValidator.java [hunk 1]
```diff
@@ -92,7 +92,9 @@ private long countMatchingGroupAttributesAssertions(final T element) {
         .filter(
             assertion -> {
               final String attributeValue = assertion.attributeSupplier.apply(element);
-              return attributeValue != null && !attributeValue.isEmpty();
+              return attributeValue != null
+                  && !attributeValue.isEmpty()
+                  && !" ".equals(attributeValue);
```
Comment (berkaycanbc) [inline bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebeElementValidator.java:97]: ❌ We should also prevent sending strings that contains multiple spaces. It is wiser to use `!attributeValue.trim().isEmpty()` in Java 8.

File: bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeUserTaskValidationTest.java [hunk 2]
```diff
@@ -77,7 +77,31 @@ public static Object[][] parameters() {
         singletonList(
             expect(
                 ZeebeFormDefinition.class,
-                "Exactly one of the attributes 'formId, formKey' must be present and not empty"))
+                "Exactly one of the attributes 'formId, formKey' must be present and not blank"))
+      },
+      {
+        Bpmn.createExecutableProcess("process")
```
Comment (berkaycanbc) [inline bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeUserTaskValidationTest.java:83]: ❌ Let's add same tests for start events as well.

**Review by berkaycanbc — CHANGES_REQUESTED** [review]
Nice work @nicpuppa! 🚀 

I left only two comments which I think is worth updating. 👍

File: bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebeElementValidator.java (hunk 1 above)
Comment (nicpuppa) [reply bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebeElementValidator.java:97]: You're right, nice one 💪

**Comment by nicpuppa** [comment]
Hey @berkaycanbc thanks for the review, I applied your suggestions. Please have another look 👀

**Review by berkaycanbc — APPROVED** [review]
LGTM! 🚀

**Comment by berkaycanbc** [comment]
bors merge

**Comment by nicpuppa** [comment]
bors retry

**Pull request merged**